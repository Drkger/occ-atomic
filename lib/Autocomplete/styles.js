import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
