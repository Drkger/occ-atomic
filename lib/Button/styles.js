import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};