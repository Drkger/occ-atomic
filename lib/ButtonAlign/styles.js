export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};