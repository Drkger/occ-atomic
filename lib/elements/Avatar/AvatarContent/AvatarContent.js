import React from 'react';
import PropTypes from 'prop-types';

const capitalLetter = (name) => {
    if (name) {
        let capitals = name.toUpperCase().split(" ");
        capitals = capitals.filter(Boolean);
        return (<span>{capitals[0] ? capitals[0].charAt(0) : 'N' }{capitals[1] ? capitals[1].charAt(0) : 'O'}</span>);
    }
    return "";
};

const AvatarContent = ({ classes, photo, gender, name, size }) => (
    <div className={classes.wrap}>
        {gender && <div className={`${classes.cont} ${classes[gender]}`} />}
        {(!gender && name) && <div className={`${classes.cont} ${classes.capital}`} style={size ? {fontSize:size*0.4, lineHeight:`${size}px`} : {}}>{capitalLetter(name)}</div>}
        {photo && <div className={classes.cont} style={{background:`url('${photo}') no-repeat center center / cover`}} />}
    </div>
);

AvatarContent.propTypes = {
    classes: PropTypes.object,
    photo: PropTypes.string,
    gender: PropTypes.string,
    name: PropTypes.string,
    size: PropTypes.number
};

export default AvatarContent;