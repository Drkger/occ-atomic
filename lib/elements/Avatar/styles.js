export default {
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
};