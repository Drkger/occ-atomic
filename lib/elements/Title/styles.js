import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';

export default {
    title: {
        boxSizing:'border-box',
        fontFamily:fonts.title,
        color:colors.title,
        fontWeight:'600',
        marginTop:'20px',
        marginBottom:'15px',
        lineHeight:'1.1'
    },
    h1: {
        fontSize:'22px'
    },
    h2: {
        fontSize:'20px',
    },
    h3: {
        fontSize:'18px',
    },
    h4: {
        fontSize:'15px'
    }
};