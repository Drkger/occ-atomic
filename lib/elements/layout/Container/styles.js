import grid from '../../../subatomic/grid';

export default {
    conFluid: {
        boxSizing:'border-box',
        paddingRight:'15px',
        paddingLeft:'15px',
        marginRight:'auto',
        marginLeft:'auto',
        '&:before, &:after': {
            display:'table',
            content:'""'
        },
        '&:after': {
            clear:'both'
        }
    },
    con: {
        extend: 'conFluid',
        margin:'0 auto'
    },
    [`@media (min-width: ${grid.sm}px)`]: {
        con: {
            width:'750px'
        }
    },
    [`@media (min-width: ${grid.md}px)`]: {
        con: {
            width:'970px'
        }
    },
    [`@media (min-width: ${grid.lg}px)`]: {
        con: {
            width:'1170px'
        }
    }
};