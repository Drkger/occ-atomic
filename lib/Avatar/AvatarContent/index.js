import injectSheet from 'react-jss';
import AvatarContent from './AvatarContent';
import styles from './styles';

export default injectSheet(styles)(AvatarContent);