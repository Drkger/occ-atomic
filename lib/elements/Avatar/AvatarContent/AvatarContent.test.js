import React from 'react';
import { shallow } from 'enzyme';
import AvatarContent from './AvatarContent';
import styles from './styles';
import AvatarContentJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("AvatarContent", () => {
    
    it('matches the snapshot', () => {
        const wrapper = shallow(
            <AvatarContent name="John Doe" classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    it('returns the first two capital letters of the name', () => {
        const wrapper = shallow(
            <AvatarContent name="John Doe" classes={classes} />
        );
        expect(wrapper.find('.capital span').text()).toEqual('JD');
    });
    
    it('shows the right gender icon', () => {
        const wrapper = shallow(
            <AvatarContent gender="m" classes={classes} />
        );
        expect(wrapper.find('.m').length).toBe(1);
        expect(wrapper.find('.f').length).toBe(0);
        wrapper.setProps({ gender: 'f' });
        expect(wrapper.find('.m').length).toBe(0);
        expect(wrapper.find('.f').length).toBe(1);
    });
    
    it('renders the photo as a background', () => {
        const wrapper = shallow(
            <AvatarContent photo="http://placehold.it/70x70" classes={classes} />
        );
        expect(wrapper.find('.cont').prop('style')).toHaveProperty('background', "url('http://placehold.it/70x70') no-repeat center center / cover")
    });
    
    it('changes the font size', () => {
        const wrapper = shallow(
            <AvatarContent name="John Doe" size={50} classes={classes} />
        );
        const styles = wrapper.find('.capital').prop('style');
        expect(styles).toHaveProperty('fontSize', '20px');
        expect(styles).toHaveProperty('lineHeight', '50px');
    });
    
});

describe("AvatarContentJSS", () => {
    
    it('matches the snapshot', () => {
        const wrapper = shallow(
            <AvatarContentJSS name="John Doe" />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe("AvatarContent styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});