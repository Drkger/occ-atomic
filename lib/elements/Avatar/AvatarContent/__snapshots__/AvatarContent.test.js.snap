// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarContent matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AvatarContent
    classes={
      Object {
        "capital": "capital",
        "cont": "cont",
        "f": "f",
        "m": "m",
        "wrap": "wrap",
      }
    }
    name="John Doe"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="cont capital"
          style={Object {}}
        >
          <span>
            J
            D
          </span>
        </div>,
        undefined,
      ],
      "className": "wrap",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            J
            D
          </span>,
          "className": "cont capital",
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "J",
              "D",
            ],
          },
          "ref": null,
          "rendered": Array [
            "J",
            "D",
          ],
          "type": "span",
        },
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="cont capital"
            style={Object {}}
          >
            <span>
              J
              D
            </span>
          </div>,
          undefined,
        ],
        "className": "wrap",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              J
              D
            </span>,
            "className": "cont capital",
            "style": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "J",
                "D",
              ],
            },
            "ref": null,
            "rendered": Array [
              "J",
              "D",
            ],
            "type": "span",
          },
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AvatarContentJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(AvatarContent)
    name="John Doe"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "capital": "AvatarContent-capital-0-1-5",
        "cont": "AvatarContent-cont-0-1-2",
        "f": "AvatarContent-f-0-1-4",
        "m": "AvatarContent-m-0-1-3",
        "wrap": "AvatarContent-wrap-0-1-1",
      },
      "name": "John Doe",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "capital": "AvatarContent-capital-0-1-5",
          "cont": "AvatarContent-cont-0-1-2",
          "f": "AvatarContent-f-0-1-4",
          "m": "AvatarContent-m-0-1-3",
          "wrap": "AvatarContent-wrap-0-1-1",
        },
        "name": "John Doe",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
